### PR TITLE
Incremental sync implementation

### PR DESCRIPTION
**Pull request by shrishrirang**
- Last major functionality of offline sync

### CODE REVIEW

**Comment by shrishrirang** [comment]
@danderson00

File: sdk/src/sync/pull.js [hunk 1]
```diff
@@ -149,19 +168,93 @@ function createPullManager(client, store, storeTaskRunner, operationTableManager
         });
     }
 
+    // Gets the last known updatedAt timestamp.
+    // For incremental pull, we check if we have any information about it in the store.
+    // If not we simply use 1970 to start the sync operation, just like a non-incremental / vanilla pull.
+    function getLastKnownUpdatedAt() {
+        
+        return Platform.async(function(callback) {
+            callback();
```
Comment (danderson00) [inline sdk/src/sync/pull.js]: Should this callback be invoked here or after obtaining the result?

Comment (shrishrirang) [inline sdk/src/sync/pull.js]: We are just starting a promise chain by creating a resolved promise. So we do intend to call the callback immediately.

Comment (danderson00) [inline sdk/src/sync/pull.js]: Ahhh, it's invoking the callback that actually resolves the promise. Got ya.

**Pull request merged**